### PR TITLE
Supermatter sliver functionality

### LINKED ISSUE #6694: Supermatter slivers get vaporized by supermatter
Linux-staging

supermatter slivers disintegrate when touched to a supermatter shard
Screenshot at 2021-06-04 20-58-08

### PR DESCRIPTION
**Pull request by ilinm1**
### Purpose
Adds functionality to SM sliver, allowing it to vaporize items and people and being put into container with tongs. Fixes #6694 

### Notes:
Since all steal objectives are appear to be broken now, sliver stealing objective may be failed too

### Changelog:
CL: [New] Supermatter sliver stealing objective
CL: [Fix] Supermatter sliver isn't getting vaporized by supermatter anymore
CL: [Improvement]  Supermatter sliver is vaporizing stuff now

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/Others/SupermatterSliver.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+		if (info.ToSlot != null && info.ToSlot?.NamedSlot != null)
+		{
+			player = info.ToRootPlayer?.PlayerScript;
+
+			if (player != null && vaporizeWhenPickedUp)
+			{
+				Chat.AddActionMsgToChat(player.PlayerChatLocation,
+					$"You reach for the {gameObject.ExpensiveName()} with your hands. That was dumb.",
+					$"{player.playerName} touches {gameObject.ExpensiveName()} with bare hands. His body bursts into flames and flashes to dust after few moments.");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Others/SupermatterSliver.cs]:
```suggestion
					$"{player.visibleName} touches {gameObject.ExpensiveName()} with bare hands. His body bursts into flames and flashes to dust after few moments.");
```

**Review by Bod9001 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Items/Others/SupermatterSliver.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	public class SupermatterSliver : MonoBehaviour, IServerInventoryMove, ICheckedInteractable<HandApply>
+	{
+		[SerializeField]
+		private ItemTrait supermatterScalpel = null;
+
+		[SerializeField]
+		private ItemTrait supermatterTongs = null;
+
+		public bool vaporizeWhenPickedUp = true;
+		private PlayerScript player;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Others/SupermatterSliver.cs]:
```suggestion
```
oh sorry i didnt actually check why codacy was complaining about this, thought it was a different issue. You dont need this.

File: UnityProject/Assets/Scripts/Items/Others/SupermatterSliver.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		}
+
+		//Turn player into ash if he picked it up
+		public void OnInventoryMoveServer(InventoryMove info)
+		{
+			if (info.InventoryMoveType != InventoryMoveType.Add) return;
+
+			if (info.ToSlot != null && info.ToSlot?.NamedSlot != null)
+			{
+				player = info.ToRootPlayer?.PlayerScript;
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Others/SupermatterSliver.cs]:
```suggestion
				var player = info.ToRootPlayer?.PlayerScript;
```

File: UnityProject/Assets/Scripts/Items/Others/SupermatterSliverContainer.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		private ItemStorage itemStorage;
+		private ItemSlot sliverSlot;
+		public bool isSealed = false;
+		public bool isLoaded = false;
+
+		private void Awake()
+		{
+			itemStorage = GetComponent<ItemStorage>();
+			sliverSlot = itemStorage.GetIndexedItemSlot(0);
+			sliverSlot.OnSlotContentsChangeServer.AddListener(OnServerSlotContentsChange);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Others/SupermatterSliverContainer.cs:25]: Can you add a remove in OnDisable for this listener, and move the AddListener to OnEnable

File: UnityProject/Assets/Scripts/Items/Tool/SupermatterTongs.cs [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+		public SpriteHandler[] spriteHandlers = new SpriteHandler[] { };
+		private ItemStorage itemStorage;
+		private ItemSlot sliverSlot;
+
+		private void Awake()
+		{
+			itemStorage = GetComponent<ItemStorage>();
+			sliverSlot = itemStorage.GetIndexedItemSlot(0);
+			sliverSlot.OnSlotContentsChangeServer.AddListener(OnServerSlotContentsChange);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Items/Tool/SupermatterTongs.cs:22]: and this one

**Comment by corp-0** [comment]
> Since all steal objectives are appear to be broken now, sliver stealing objective may be failed too

are they? What's wrong with them?

**Comment by ilinm1** [comment]
#7814 not sure if thats true for all objectives but steal objectives are failing often

File: UnityProject/Assets/Scripts/Items/Tool/SupermatterTongs.cs (hunk 5 above)
Comment (ilinm1) [reply UnityProject/Assets/Scripts/Items/Tool/SupermatterTongs.cs:22]: Fixed

File: UnityProject/Assets/Scripts/Items/Others/SupermatterSliverContainer.cs (hunk 4 above)
Comment (ilinm1) [reply UnityProject/Assets/Scripts/Items/Others/SupermatterSliverContainer.cs:25]: Fixed

**Review by ThatDan123 — APPROVED** [review]

**Review by NoooneyDude — APPROVED** [review]

**Pull request merged**